### PR TITLE
Add metrics and alerts to NFD Operator

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This patch enables NFD Operator to expose metrics about its health (NFD master and worker daemonset health)

### CODE REVIEW

File: config/crd/bases/nfd.openshift.io_nodefeaturediscoveries.yaml [hunk 1]
```diff
@@ -74,12 +105,42 @@ spec:
...
+                          - type
+                          type: object
+                        type: array
+                    type: object
                 required:
                 - configData
                 type: object
             required:
             - operand
-            - workerConfig
```
Comment (dagrayvid) [inline config/crd/bases/nfd.openshift.io_nodefeaturediscoveries.yaml:82]: Is this necessary?

File: config/default/manager_config_patch.yaml [hunk 2]
```diff
@@ -8,14 +8,16 @@ spec:
...
+          readOnly: true
       volumes:
-      - name: manager-config
-        configMap:
-          name: manager-config
-
+      - name: cert
+        secret:
+          defaultMode: 420
+          secretName: webhook-server-cert
```
Comment (dagrayvid) [inline config/default/manager_config_patch.yaml:23]: Add newline?

File: config/crd/bases/nfd.openshift.io_nodefeaturediscoveries.yaml (hunk 1 above)
Comment (ArangoGutierrez) [reply config/crd/bases/nfd.openshift.io_nodefeaturediscoveries.yaml:82]: not at all

File: config/default/manager_config_patch.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply config/default/manager_config_patch.yaml:23]: ups

File: manifests/4.9/bundle.Dockerfile [hunk 3]
```diff
@@ -7,10 +7,15 @@ LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
...
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.4.0+git
+
+# Labels for testing.
+LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
 
 # Copy files to locations specified by labels.
 COPY manifests /manifests/
 COPY metadata /metadata/
+COPY tests/scorecard /tests/scorecard/
```
Comment (dagrayvid) [inline manifests/4.9/bundle.Dockerfile:21]: @ArangoGutierrez

File: pkg/config/config.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+package config
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	nodeFeautreDiscoveryImageDefault string = "quay.io/openshift/origin-node-feature-discovery:4.9"
```
Comment (dagrayvid) [inline pkg/config/config.go:25]:
```suggestion
	nodeFeatureDiscoveryImageDefault string = "quay.io/openshift/origin-node-feature-discovery:4.9"
```

File: pkg/config/config.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+// NodeFeatureDiscoveryImage returns the operator's operand/nfd image.
+func NodeFeatureDiscoveryImage() string {
+	nodeFeatureDiscoveryImage := os.Getenv("NODE_FEATURE_DISCOVERY_IMAGE")
+
+	if len(nodeFeatureDiscoveryImage) > 0 {
+		return nodeFeatureDiscoveryImage
+	}
+
+	return nodeFeautreDiscoveryImageDefault
```
Comment (dagrayvid) [inline pkg/config/config.go:73]:
```suggestion
	return nodeFeatureDiscoveryImageDefault
```

File: version/version.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+var (
+	Version = "0.0.1"
```
Comment (dagrayvid) [inline version/version.go:19]:
```suggestion
	Version = "4.9.0"
```
Or something else?

**Pull request merged**